tests/datasets/test_dataloader.py

```diff
@@ -263,3 +263,39 @@ def test_subindicators_update(self, dataset, datasetData, subindicatorGroup):
         assert len(groups) == 2
         assert groups[0].subindicators == ["A", "B", "C"]
         assert groups[1].subindicators == ["X", "Y"]
+
+    def test_new_dataset_same_group_name(self, dataset, datasetData, subindicatorGroup):
+        # Make sure datasets with new subindicators are not leaking into groups with the same name from previous datasets
+        new_dataset = DatasetFactory()
+
+        DatasetDataFactory(
+            dataset=new_dataset, data={
+                'group1': 'D', 'group2': 'Z'
+            }
+        )
+
+        groups = dataloader.create_groups(new_dataset, ["group1", "group2"])
+
+        assert len(groups) == 2
+        assert groups[0].subindicators == ["D"]
+        assert groups[0].subindicators != ["A", "B", "C", "D"]
```

Comment:
not quite right. 

`datasetData` creates two subindicators `A` and `B` but not `C` so this will never be the case.